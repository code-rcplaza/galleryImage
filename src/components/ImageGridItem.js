import React from 'react'
import PropTypes from 'prop-types'
import { capitalizeFirstLetter } from '../helpers/capitalizeFirstLetter'

const ImageGridItem = ({ title, url }) => {
    
    return (
        <a href={url} rel="noopener noreferrer" target="_blank">
            <div className="img-container animate__animated animate__fadeIn animate__delay-0.5s">
                <img className="img-item" src={url} alt="img" />
                <div className="line-container">
                    <span className="single-line">{capitalizeFirstLetter(title)}</span>
                </div>
            </div>
        </a>
    )
}


ImageGridItem.propTypes = {
    url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
}

export default ImageGridItem

