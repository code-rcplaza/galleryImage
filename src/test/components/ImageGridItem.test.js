import React from 'react';
import { shallow } from 'enzyme'
import ImageGridItem from '../../components/ImageGridItem';

describe('Unite test to <ImageGridItem />', () => {
    
    const title = 'Gif Title'
    const url = 'https://giphy.com/gifs/rurouni-kenshin-samurai-x-eo8UrZwpvjPHi'

    let wrapper = shallow(<ImageGridItem title={ title } url= { url }/>)
    
    beforeEach( () => {
        wrapper = shallow(<ImageGridItem title={ title } url= { url } />)
    })

    test('Correctly Render', () => {
        expect( wrapper ).toMatchSnapshot()
    })

    test('Title props is string', () => {
        const titleText = wrapper.find('span').text().trim()

        expect( titleText ).toBe( title )
    })

    test('URL and alt exist in img element', () => {
        const img = wrapper.find('img')

        expect( img.prop('src') ).toBe( url )
        expect( img.prop('alt') ).toBe( 'img' )
    })
})