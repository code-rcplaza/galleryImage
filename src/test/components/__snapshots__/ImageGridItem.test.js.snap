// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unite test to <ImageGridItem /> Correctly Render 1`] = `
<a
  href="https://giphy.com/gifs/rurouni-kenshin-samurai-x-eo8UrZwpvjPHi"
  rel="noopener noreferrer"
  target="_blank"
>
  <div
    className="img-container animate__animated animate__fadeIn animate__delay-0.5s"
  >
    <img
      alt="img"
      className="img-item"
      src="https://giphy.com/gifs/rurouni-kenshin-samurai-x-eo8UrZwpvjPHi"
    />
    <div
      className="line-container"
    >
      <span
        className="single-line"
      >
        Gif Title
      </span>
    </div>
  </div>
</a>
`;
